### DISCUSSION TITLE
Why is Dynamic Link Variable not interpolated on Link => getServerSideProps

### DISCUSSION THREAD

**Discussion by followbl** [vercel/next.js#14822] [Help] [answered]
In the below example - if I hit the url directly `https://www.example.com/post/abc` page works fine (context.params.slug) is fine...but if I navigate from a Next.js link from a feed page as shown below - params (and query) are not interpolated, but returned as `[slug]`, not the abc I'm looking for. 

On top of this - this only happens in the production build deployed to vercel - this behavior does not happen in development...

something I'm doing wrong?

Page: https://www.example.com/feed
```tsx
<Link href="/post/[slug]" as="/post/abc">
  <a>First Post</a>
</Link>
```

Page: https://www.example.com/post/[slug]
```tsx
export const getServerSideProps: GetServerSideProps = async (context) => {
  const postData = await fetcher(`/api/${context.params.slug}`, context.req.headers);

  return {
    props: {
      initialData: !!postData ? postData : null,
    },
  };
};

**Comment by followbl** [comment]
For more context - here is straight from the Vercel deployed logs

<img width="1413" alt="Screen Shot 2020-07-03 at 11 47 32 AM" src="https://user-images.githubusercontent.com/927065/86483941-1f02be80-bd23-11ea-9673-77c620619988.png">

**Comment by jamesmosier** [comment]
Hey @followbl! Does this work as intended if you build/start locally? For me it does, but I just wanted to narrow it down a bit.

**Reply by followbl** [reply]
@jamesmosier yes works locally, not in prod - (buried in my long-winded comment above I mentioned it)

**Comment by timneutkens** [comment, marked as answer]
Thanks for the report! Had @ijjk look into this, here's the PR to fix it: https://github.com/vercel/vercel/pull/4778

**Reply by followbl** [reply]
@timneutkens least I can do. Thank you and the Next.js team for all your work on an amazing piece of software.
